src/StringTools.UnitTests/OpportunisticIntern_Tests.cs

```diff
@@ -0,0 +1,142 @@
+
+#if NET35_UNITTEST
+extern alias StringToolsNet35;
+#endif
+
+using Shouldly;
+using Xunit;
+
+#if NET35_UNITTEST
+using StringToolsNet35::System;
+using StringToolsNet35::Microsoft.StringTools;
+#else
+using System;
+#endif
+
+namespace Microsoft.StringTools.Tests
+{
+    public class OpportunisticInternTestBase
+    {
+        private static bool IsInternable(ref InternableString internable)
+        {
+            string i1 = OpportunisticIntern.Instance.InternableToString(ref internable);
+            string i2 = OpportunisticIntern.Instance.InternableToString(ref internable);
+            i1.ShouldBe(i2); // No matter what, the same string value should return.
+            return System.Object.ReferenceEquals(i1, i2);
+        }
+
+        private static void AssertInternable(ref InternableString internable)
+        {
+            IsInternable(ref internable).ShouldBeTrue();
+        }
+
+        private static string AssertInternable(char[] ch, int startIndex, int count)
+        {
+            var target = new InternableString(ch.AsSpan(startIndex, count));
+            AssertInternable(ref target);
+            target.Length.ShouldBe(count);
+
+            return target.ExpensiveConvertToString();
+        }
+
+        private static void AssertInternable(string value)
+        {
+            AssertInternable(value.ToCharArray(), 0, value.ToCharArray().Length);
+        }
+
+        private static void AssertNotInternable(InternableString internable)
+        {
+            IsInternable(ref internable).ShouldBeFalse();
+        }
+
+        private static void AssertNotInternable(char[] ch)
+        {
+            AssertNotInternable(new InternableString(ch.AsSpan(0, ch.Length)));
+        }
+
+        protected static void AssertNotInternable(string value)
+        {
+            AssertNotInternable(value.ToCharArray());
+        }
+
+        /// <summary>
+        /// Test interning segment of char array
+        /// </summary>
+        [Fact]
+        public void SubArray()
+        {
+            var result = AssertInternable(new char[] { 'a', 't', 'r', 'u', 'e' }, 1, 4);
+            result.ShouldBe("true");
+        }
+
+        /// <summary>
+        /// Test interning segment of char array
+        /// </summary>
+        [Fact]
+        public void SubArray2()
+        {
+            var result = AssertInternable(new char[] { 'a', 't', 'r', 'u', 'e', 'x' }, 1, 4);
+            result.ShouldBe("true");
+        }
+
+        /// <summary>
+        /// This is the list of hard-coded interns. They should report interned even though they are too small for normal interning.
+        /// </summary>
+        [Fact]
+        public void KnownInternableTinyStrings()
+        {
+            AssertInternable("C#");
+            AssertInternable("F#");
+            AssertInternable("VB");
+            AssertInternable("True");
+            AssertInternable("TRUE");
```

Comment:
Is there a way to ensure they go down the hardcoded path?

Reply:
We could tell by reading the interning stats but I'll first verify that there is value to hardcoding at all. Thanks.

Reply:
I have completely deleted this file, together with the hardcoded string support.